src/amo/reducers/recommendations.js

```diff
@@ -0,0 +1,182 @@
+/* @flow */
+import invariant from 'invariant';
+
+import { createInternalAddon } from 'core/reducers/addons';
+import type { ExternalAddonType } from 'core/types/addons';
+
+export const ABORT_FETCH_RECOMMENDATIONS: 'ABORT_FETCH_RECOMMENDATIONS'
+  = 'ABORT_FETCH_RECOMMENDATIONS';
+export const FETCH_RECOMMENDATIONS: 'FETCH_RECOMMENDATIONS'
+  = 'FETCH_RECOMMENDATIONS';
+export const LOAD_RECOMMENDATIONS: 'LOAD_RECOMMENDATIONS'
+  = 'LOAD_RECOMMENDATIONS';
+
+export type Recommendations = {|
+  addons: Array<ExternalAddonType> | null,
+  fallbackReason: string | null,
+  loading: boolean,
+  outcome: string | null,
```

Comment:
Do we want to specify the allowed values? [The API documentation](http://addons-server.readthedocs.io/en/latest/topics/api/addons.html#recommendations) describes the following possible values:

> outcome (string) – Outcome of the response returned. Will be either: recommended - responses from recommendation service; recommended_fallback - service timed out or returned empty results so we returned fallback; curated - recommended=False was requested so fallback returned.

Reply:
Because the app doesn't care what the value is (i.e., all it does is record the value, but it never makes any decisions based on the value), I thought it might make sense to leave this open. Theoretically the API could change the strings, or add a string, and it shouldn't break anything.

However this does seem to go against our use of Flow, so I will update this to a type which allows specific strings, and I'll do the same for `fallbackReason`.